client/components/editor/teaching-elements/Assessment/TextSelection/helpers.js

```diff
@@ -0,0 +1,81 @@
+import Blast from 'blast-vanilla';
+import forEach from 'lodash/forEach';
+import generateRange from 'lodash/range';
+
+const blastClass = 'blast';
+const idPrefix = 'text-';
+
+export function generateHtml(content, iteratee) {
+  if (!content) return '';
+  const container = document.createElement('div');
+  container.innerHTML = content;
+  // eslint-disable-next-line no-new
+  new Blast(container);
+  const nodes = filterNodes(container, NodeFilter.SHOW_TEXT);
+  nodes.forEach(iteratee);
+  return container.innerHTML;
+}
+
+export function isBlast(node) {
+  if (!(node instanceof HTMLSpanElement)) return false;
+  return node.classList.contains(blastClass);
+}
+
+export function getText(element, range) {
+  const query = index => `span#${idPrefix}${index}.${blastClass}`;
+  const indexes = generateRange(...range);
+  return indexes.map(index => element.querySelector(query(index)));
+}
+
+export function toggleAttributes(element, config) {
+  forEach(config, (value, name) => {
+    const action = value !== null ? 'setAttribute' : 'removeAttribute';
+    element[action](name, value);
+  });
+}
+
+export function toggleClasses(element, config) {
+  forEach(config, (value, name) => {
+    element.classList.toggle(name, value);
+  });
+}
+
+export function processNode(node, selections) {
+  const index = extractIndex(node);
+  const selection = selections.getByIndex(index);
+  if (!selection) return node;
+  node.classList.add('selected');
+  node.setAttribute('range', [...selection]);
```

Comment:
If this is why you opted to implementing iterator on `Range` you could simply done:
```js
node.setAttribute('range', [selection.start, selection.end]);
```
And btw. this is not **selection** but **range** 😉